lib/IRGen/GenStruct.cpp

```diff
@@ -879,19 +879,29 @@ void IRGenModule::emitStructDecl(StructDecl *st) {
   emitNestedTypeDecls(st->getMembers());
 }
 
+void IRGenModule::maybeEmitOpaqueTypeDecl(OpaqueTypeDecl *opaque) {
+  if (IRGen.Opts.EnableAnonymousContextMangledNames) {
```

Comment:
Perhaps fold this check into hasLazyMetadata()?

Reply:
I'm not sure that buys much, since there's not much to gain from putting these in RuntimeResolvableTypes if we aren't also putting the name info in anonymous contexts, so the condition would be still be here, and still be worth factoring out of the individual `emit` methods below.